datadog_checks_base/datadog_checks/base/checks/win/wmi/sampler.py

```diff
@@ -142,7 +143,28 @@ def __init__(
         self.filters = filters
         self._and_props = and_props if and_props is not None else []
         self._timeout_duration = timeout_duration
-        self._query = timeout(timeout_duration)(self._query)
+
+        self._runSampleEvent = Event()
+        self._sampleComplete = Event()
+        self.setDaemon(True)
+
+        self.start()
+
+    def run(self):
+        try:
+            pythoncom.CoInitialize()
+        except Exception as e:
+            self.logger.info("exception in CoInitialize {}".format(e))
+            raise
+
+        while True:
+            self._runSampleEvent.wait()
+            if self.is_raw_perf_class and not self._previous_sample:
+                self._current_sample = self._query()
+
+            self._previous_sample = self._current_sample
+            self._current_sample = self._query()
```

Comment:
Unless I'm being daft, `self._query` is never defined

Reply:
I think it is defined [here on line 433](https://github.com/DataDog/integrations-core/blob/322eafd203775b9e39e4561e3ff6647b2e6c6d14/datadog_checks_base/datadog_checks/base/checks/win/wmi/sampler.py#L433)

Reply:
Ah ty, I was misreading the deletion on line 145